pkg/hive/install.go

```diff
@@ -0,0 +1,211 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+	"github.com/Azure/ARO-RP/pkg/util/steps"
+	"github.com/Azure/ARO-RP/pkg/util/version"
+)
+
+const (
+	argSecretsName       = "aro-env-secret"
+	pullsecretSecretName = "aro-pullsecret"
+	installConfigName    = "aro-installconfig"
+)
+
+func makeEnvSecret(name string) corev1.EnvVar {
+	return corev1.EnvVar{
+		Name: "ARO_" + name,
+		ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: argSecretsName,
+				},
+				Key: "ARO_" + name,
+			},
+		},
+	}
+}
+
+func (c *clusterManager) Install(ctx context.Context, oc *api.OpenShiftCluster, sub *api.Subscription) error {
+	s := []steps.Step{
+		steps.Action(func(ctx context.Context) error { return c.installViaHive(ctx, oc, sub) }),
+		steps.Condition(c.waitForHive, 30*time.Minute, true),
+	}
+
+	err := steps.Run(ctx, c.log, 10*time.Second, s)
+	return err
+}
+
+func (hr *clusterManager) installViaHive(ctx context.Context, oc *api.OpenShiftCluster, sub *api.Subscription) error {
+	sppSecret, err := hr.servicePrincipalSecretForInstall(oc, sub)
+	if err != nil {
+		return err
+	}
+
+	psSecret, err := pullsecretSecret(hr.namespaceName(), oc)
+	if err != nil {
+		return err
+	}
+
+	resources := []kruntime.Object{
+		hiveNamespace(hr.namespaceName()),
+		sppSecret,
+		envSecret(hr.namespaceName(), hr.env.IsLocalDevelopmentMode()),
+		psSecret,
+		installConfigCM(hr.namespaceName(), oc.Location),
+		hr.clusterDeploymentForInstall(oc, hr.env.IsLocalDevelopmentMode()),
+	}
+
+	err = dynamichelper.Prepare(resources)
+	if err != nil {
+		return err
+	}
+
+	err = hr.dh.Ensure(ctx, resources...)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (hr *clusterManager) servicePrincipalSecretForInstall(oc *api.OpenShiftCluster, sub *api.Subscription) (*corev1.Secret, error) {
+	clusterSPBytes, err := hr.generateSPBytes(oc, sub)
+	if err != nil {
+		return nil, err
+	}
+
+	enc, err := json.Marshal(oc)
+	if err != nil {
+		return nil, err
+	}
+
+	encSub, err := json.Marshal(sub)
+	if err != nil {
+		return nil, err
+	}
+
+	proxyCert, err := ioutil.ReadFile("secrets/proxy.crt")
+	if err != nil {
+		return nil, err
+	}
+
+	proxyClientCert, err := ioutil.ReadFile("secrets/proxy-client.crt")
+	if err != nil {
+		return nil, err
+	}
+
+	proxyClientKey, err := ioutil.ReadFile("secrets/proxy-client.key")
+	if err != nil {
+		return nil, err
+	}
+
+	sppSecret := servicePrincipalSecret(hr.namespaceName(), clusterSPBytes)
+
+	data := map[string][]byte{
+		"osServicePrincipal.json": sppSecret.Data["osServicePrincipal.json"],
+		"99_aro.json":             enc,
+		"99_sub.json":             encSub,
+		"proxy.crt":               proxyCert,
+		"proxy-client.crt":        proxyClientCert,
+		"proxy-client.key":        proxyClientKey,
```

Comment:
No, this is for the Installer, in case your Hive is local (e.g. microk8s or whatever). Getting rid of this would mean that we would be reliant on the development AKSes with no local option.